docs/modules/ROOT/pages/tmail-backend/jmap-extensions/pushWithFirebase.adoc

```diff
@@ -0,0 +1,210 @@
+
+1. A unique identifier associated with the device where the JMAP client is
+running, normally supplied by the device's operating system.
+2. A custom vendor/app id, including a domain controlled by the vendor of
+the JMAP client.
+
+    To protect the privacy of the user, the deviceClientId id MUST NOT contain
+    an unobfuscated device id.
+
+- **expires**: `UTCDate|null`
+  The time this push subscription expires. If specified, the JMAP server MUST NOT make further requests to this resource after this time. It MAY automatically destroy the push subscription at or after this time.
+
+    The server MAY choose to set an expiry if none is given by the client or modify the expiry time given by the client to a shorter duration.
+
+- **types**: `String[]|null`
+A list of types the client is interested in (using the same names as the keys in the *TypeState* object defined in the previous section). A StateChange notification will only be sent if the data for one of these types changes. Other types are omitted from the TypeState object. If `null`, changes will be pushed for all types.
+
+=== FirebaseRegistration/get
+
+Standard */get* method as described in Section 5.1, except it does **not** take or return an *accountId* argument,
+as push subscriptions are not tied to specific accounts. It also does **not** return a *state* argument. The *ids*
+argument may be `null` to fetch all at once.
+
+It does **not** return a *state* argument.
+
+As the *token* properties may contain data that is private to a particular device, the values for these properties MUST NOT be returned.
+
+=== FirebaseRegistration/set
+
+Standard */set* method  as described in Section 5.3, except it does **not** take or return an *accountId* argument,
+as push subscriptions are not tied to specific accounts. It also does **not** take an *ifInState* argument or return
+*oldState* or *newState* arguments.
+
+The client may update the *expires* property to extend (or, less commonly, shorten) the lifetime of a push subscription.
+The server MAY modify the proposed new expiry time to enforce server-defined limits.
+
+Clients SHOULD NOT update or destroy a push subscription that they did not create (i.e., has a *deviceClientId* that
+they do not recognise).
+
+=== Example
+
+At `2018-07-06T02:14:29Z`, a client with deviceClientId `a889-ffea-910` fetches the set of FirebaseRegistration
+currently on the server, making an API request with:
+
+....
+    [[ "FirebaseRegistration/get", {
+      "ids": null
+    }, "0" ]]
+....
+
+Which returns:
+
+....
+    [[ "FirebaseRegistration/get", {
+      "list": [{
+          "id": "e50b2c1d-9553-41a3-b0a7-a7d26b599ee1",
```

Comment:
Do we need a `active` status for case client register to James, but James not yet register to firebase?
eg: waiting, failed, success

Reply:
no